lib/platform/azure/index.ts

```diff
@@ -149,12 +153,13 @@ export async function initRepo({
     url: defaults.endpoint,
   });
   const url =
```

Comment:
```suggestion
  const parsedEndpoint = URL.parse(defaults.endpoint +
      `${encodeURIComponent(projectName)}/_git/${encodeURIComponent(repoName)}`);
  if(opts.Token) {
    parsedEndpoint.auth = token.length === 52 ? `:${opts.token}` : `token:${opts.token}`
  } else {
    parsedEndpoint.auth = `${opts.username}:${opts.password}`; 
  }
  const url = URL.format(parsedEndpoint);
```

So we don't need extra clone opts here

Reply:
@viceice , as mentioned in the description and in a previous iteration (cfr [here](https://github.com/renovatebot/renovate/pull/5602#discussion_r386097520))

- via clone opts was the only way to make it work consistently for TFS/Azure DevOps Server/TFS and Azure DevOps Services. Putting it in the URL does not work on our side (Azure DevOps Server) and does not work for bearer token.
- personal access token has token in its name, having people put it in a password when there is a token field is confusing. And more importantly this means it is a breaking change, for example the [azure extension](https://marketplace.visualstudio.com/items?itemName=jyc.vsts-extensions-renovate-me) would break.